src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -87,6 +88,32 @@ class BasePaymentsPage extends React.Component {
             this.debounceSetShouldShowLoadingSpinner();
         }
 
+        if (this.state.shouldShowDefaultDeleteMenu || this.state.shouldShowPasswordPrompt) {
+            // We should reset selected payment method state values and close corresponding modals if the selected payment method is deleted
+            let shouldResetPaymentMethodData = false;
+
+            if (this.state.selectedPaymentMethodType === CONST.PAYMENT_METHODS.BANK_ACCOUNT && _.isEmpty(this.props.bankAccountList[this.state.methodID])) {
+                // If the selected payment method is bank account and it has been deleted, reset the selected payment method data and close corresponding modals
```

Comment:
I think we should remove the comments from this if else chain (lines 96, 99 and 102) bc they explain what the code is doing instead of why and it is easy to understand the logic without them

Reply:
Nice! Is that for you @abdulrahuman5196?

Reply:
removed the comment now.